### PR TITLE
12 Hour User-Guild Cooldown, Games immediately reset

### PR DESCRIPTION
**Pull request by MatthewMerrill**
Example with 2 minute cooldown:

image

### CODE REVIEW

File: src/main/java/io/chrislowe/discordle/Main.java [hunk 1]
```diff
@@ -189,6 +188,7 @@ public Mono<Void> handleInteractionEvent(ChatInputInteractionEvent event, Gatewa
                     case ACCEPTED, GAME_WON, GAME_LOST -> null;
                     case INVALID_WORD -> "Your word is not in the dictionary.";
                     case GAME_UNAVAILABLE -> "There is currently no game going on. Games begin at 12AM/12PM PST.";
```
Comment (HelloChrisLowe) [inline src/main/java/io/chrislowe/discordle/Main.java]: I think this case should be obsolete now

File: src/main/java/io/chrislowe/discordle/database/repository/GameMoveRepository.java [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
 import io.chrislowe.discordle.database.dbo.GameMove;
+import io.chrislowe.discordle.database.dbo.Guild;
+import io.chrislowe.discordle.database.dbo.User;
 import org.springframework.data.repository.CrudRepository;
 
+import java.util.Optional;
+
 public interface GameMoveRepository extends CrudRepository<GameMove, Integer> {
 
+  Optional<GameMove> findTopByGame_GuildAndUserOrderByDatetimeCreatedDesc(
```
Comment (HelloChrisLowe) [inline src/main/java/io/chrislowe/discordle/database/repository/GameMoveRepository.java]: Is the _ necessary for the spring data parsing?

File: src/main/java/io/chrislowe/discordle/game/GameService.java [hunk 3]
```diff
@@ -36,6 +40,8 @@ public SubmissionOutcome submitGuess(String guildId, String discordId, String gu
             return SubmissionOutcome.GAME_UNAVAILABLE;
```
Comment (HelloChrisLowe) [inline src/main/java/io/chrislowe/discordle/game/GameService.java]: I think now with this change that the game being null should be an error

**Review by HelloChrisLowe — CHANGES_REQUESTED** [review]
Cool change, and thanks for setting up that test!

File: src/main/java/io/chrislowe/discordle/game/GameService.java (hunk 3 above)
Comment (MatthewMerrill) [reply src/main/java/io/chrislowe/discordle/game/GameService.java]: Done

File: src/main/java/io/chrislowe/discordle/database/repository/GameMoveRepository.java (hunk 2 above)
Comment (MatthewMerrill) [reply src/main/java/io/chrislowe/discordle/database/repository/GameMoveRepository.java]: Done

File: src/main/java/io/chrislowe/discordle/Main.java (hunk 1 above)
Comment (MatthewMerrill) [reply src/main/java/io/chrislowe/discordle/Main.java]: Done

**Comment by MatthewMerrill** [comment]
Done done and done

**Review by HelloChrisLowe — APPROVED** [review]

**Comment by HelloChrisLowe** [comment]
Thanks for the changes!

**Pull request merged**